### PR TITLE
Stop publishing liquibase-extension-testing to Github on branch builds (DAT-13038)

### PR DESCRIPTION
**Pull request by abrackx**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description
Keeps track of liquibase-extension-testing in cleanup jobs, publishes package on release instead.  

## Things to be aware of
This will currently publish liquibase-core and liquibase-extension-testing on release to github packages.

### CODE REVIEW

File: .github/workflows/release-published.yml [hunk 1]
```diff
@@ -195,3 +195,31 @@ jobs:
...
+          distribution: 'temurin'
+          cache: 'maven'
+          server-id: github
+
+      - name: Version artifact
+        run: mvn versions:set -DnewVersion="${{ needs.setup.outputs.version }}"
+
+      # Publish to GitHub Packages
+      - name: Publish package to Github
+        run: mvn -B clean deploy -DskipTests=true
```
Comment (StevenMassaro) [inline .github/workflows/release-published.yml:222]: Is Maven configured to only deploy the modules we care about to GitHub packages? I am probably missing that somewhere.

**Review by StevenMassaro — APPROVED** [review]

File: .github/workflows/cleanup-branch-builds.yml [hunk 2]
```diff
@@ -9,6 +9,10 @@ jobs:
   delete-package:
     name: Delete Github Package for Branch
     runs-on: ubuntu-20.04
+    matrix:
+      package:
+        - org.liquibase.liquibase-core
+        - org.liquibase.liquibase-extension-testing
```
Comment (jnewton03) [inline .github/workflows/cleanup-branch-builds.yml:15]: Is deleting branch builds needed if we aren't publishing branch builds?

Comment (abrackx) [reply .github/workflows/cleanup-branch-builds.yml]: Good point, removed!

File: .github/workflows/release-published.yml (hunk 1 above)
Comment (abrackx) [reply .github/workflows/release-published.yml:222]: Yeah we have `<maven.deploy.skip>false</maven.deploy.skip>` set in the pom files of all submodules (except core), but in liquibase-extension-testing it is set to `true` in the profile `github-package` when maven detects the env var `TARGET_ARTIFACT_REPOSITORY` is set to `github`. This deploy step is separate from the one that actually pushes to sonatype (which does not set the env var), so in this new step we'll be pushing the core artifact and extension-testing artifact on release only to github packages.

**Review by jnewton03 — APPROVED** [review]
LGTM 🚀

**Review by filipelautert — APPROVED** [review]

**Pull request merged**